Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -756,6 +769,18 @@ extension ByteBufferAllocator {
         return buffer
     }
 
+    /// Create a fresh `ByteBuffer` containing the `bytes` decoded from the ASCII `hexEncodedBytes` string .
+    ///
+    /// This will allocate a new `ByteBuffer` with enough space to fit `bytes` and potentially some extra space.
+    ///
+    /// - returns: The `ByteBuffer` containing the written bytes.
+    @inlinable
+    public func buffer(hexEncodedBytes string: String) -> ByteBuffer {
```

Comment:
This is an uncommon API shape in NIO, this should be spelled as an `init`ialiser, no?

Reply:
String, substring all have the same methods in ByteBufferAllocator. I think the initialiser you are looking for is below. 
I think this is following the pattern of the repo. 

![Screenshot 2024-08-19 at 1 29 41 PM](https://github.com/user-attachments/assets/971a1815-5280-4a08-ae32-fb5c9f5afd38)

Reply:
Yeah, this is the standard shape for ByteBufferAllocator. While I agree with you that in hindsight this was the _wrong_ API shape, it's where we are now and the consistency does matter.